internals/src/num/mod.rs

```diff
@@ -0,0 +1,23 @@
+//! Helpers for integers
+
+/// Extensions for signed integers
+pub trait IntExt {
+    /// the unsigned integer type
+    type Unsigned;
+
+    ///use this to find unsigned absolute if rust version less that 1.51
+    #[cfg(not(rust_v_1_51))]
+    fn unsigned_abs(self) -> Self::Unsigned;
+}
+
+impl IntExt for i64 {
+    type Unsigned = u64;
+
+    fn unsigned_abs(self) -> Self::Unsigned {
```

Comment:
Needs a feature gate too.

Reply:
Ouch, why didn't CI catch this?

Reply:
WTF, that's really weird.

Reply:
Ok, prior to #1608, `rust_v_1_53` was a *feature flag* not a cfg parameter, so the gate was incorrect, and `not(rust_v_1_53)` would have always been true. On Rust 1.53 this *would've* caused a warning, but this PR suppressed it, something we mistakenly thought was needed for the case that people compile on 1.53 without the feature flag on.

After #1608 this would start causing compilation errors on 1.51 and above.... **except** that the flag has always been rust_v_1_**53** while this PR uses rust_v_1_**51**.

So, multiple mistakes and multiple review failures there. Oops.

Reply:
51 is correct since that's the version the method was added in, we just need to add it into `build.rs`.